### PR TITLE
Nanos cleanup

### PR DESCRIPTION
**Pull request by kennyqiyenkan**
Removed unnecessary TypeCasting and usage of MapDAO and AbstractDAO

### CODE REVIEW

File: src/foam/nanos/boot/Boot.java [hunk 1]
```diff
@@ -17,30 +17,33 @@
...
-  protected X      root_ = new ProxyX();
+  protected DAO    pmDAO_;
+  protected X         root_ = new ProxyX();
 
   public Boot() {
     // Used for all the services that will be required when Booting
-    serviceDAO_ = new MapDAO();
-    ((MapDAO) serviceDAO_).setOf(NSpec.getOwnClassInfo());
-    ((MapDAO) serviceDAO_).setX(root_);
+    MapDAO serviceDAOConstruct = new MapDAO();
```
Comment (kgrgreer) [inline src/foam/nanos/boot/Boot.java]: Just rename to serviceDAO. serviceDAOConstruct is a little long.

File: src/foam/nanos/boot/Boot.java [hunk 2]
```diff
@@ -17,30 +17,33 @@
...
+    serviceDAOConstruct.setOf(NSpec.getOwnClassInfo());
+    serviceDAOConstruct.setX(root_);
+    serviceDAO_ = serviceDAOConstruct;
 
     // Used to hold all of the users in our system
-    userDAO_ = new MapDAO();
-    ((MapDAO) userDAO_).setOf(User.getOwnClassInfo());
-    ((MapDAO) userDAO_).setX(root_);
+    MapDAO userDAOConstruct = new MapDAO();
```
Comment (kgrgreer) [inline src/foam/nanos/boot/Boot.java]: See above.

File: src/foam/nanos/boot/Boot.java [hunk 3]
```diff
@@ -17,30 +17,33 @@
...
+    userDAOConstruct.setOf(User.getOwnClassInfo());
+    userDAOConstruct.setX(root_);
+    userDAO_ = userDAOConstruct;
     root_.put("userDAO", userDAO_);
 
     // Used for groups. We have multiple groups that contain different users
-    groupDAO_ = new MapDAO();
-    ((MapDAO) groupDAO_).setOf(Group.getOwnClassInfo());
-    ((MapDAO) groupDAO_).setX(root_);
+    MapDAO groupDAOConstruct = new MapDAO();
```
Comment (kgrgreer) [inline src/foam/nanos/boot/Boot.java]: See above.

File: src/foam/nanos/boot/Boot.java (hunk 1 above)
Comment (kennyqiyenkan) [reply src/foam/nanos/boot/Boot.java]: fair enough.

**Pull request merged**